src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -906,14 +904,19 @@ impl ThinPool {
                     let data_segments =
                         data_segment.map(|seg| coalesce_segs(data_existing_segments, &[seg]));
                     if let Some(mut ds) = data_segments {
+                        thinpooldev.suspend(get_dm(), DmOptions::default())?;
                         // Leaves data device suspended
-                        thinpooldev.set_data_table(get_dm(), segs_to_table(device, &ds))?;
+                        let res = thinpooldev.set_data_table(get_dm(), segs_to_table(device, &ds));
```

Comment:
Might as well drop the comment just above. I think it was a note on operation, now obsolete.